pkg/core/system_names/system_names.go

```diff
@@ -1,15 +1,37 @@
 package system_names
 
 import (
+	"regexp"
 	"strings"
 )
 
 const SystemPrefix = "system_"
 
+const separator = "_"
+
+var cleanNameRegex = regexp.MustCompile(`[^a-z0-9-_]`)
```

Comment:
Should we clean the name or just name it correctly and validate that we set it based on the MADR?